components/PillAction.tsx

```diff
@@ -20,14 +20,34 @@ export default function Pill({
   icon?: keyof typeof icons | undefined;
   onClick?: () => void;
 }) {
+  const [pillActionIconName, setPillActionIconName] = useState(icon || 'copy-to-clipboard');
+  const handleClick = () => {
+    if (onClick) onClick();
+    if (icon == 'copy-to-clipboard') setPillActionIconName('copy-to-clipboard-succeeded');
+  };
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+
+    if (pillActionIconName === 'copy-to-clipboard-succeeded') {
+      timeoutId = setTimeout(() => {
+        setPillActionIconName('copy-to-clipboard');
+      }, 2000);
+    }
+
+    return () => {
+      if (timeoutId) clearTimeout(timeoutId);
+    };
+  }, [pillActionIconName]);
```

Comment:
Instead of using `useEffect` could you do something like the below.

```
    if (icon == 'copy-to-clipboard') {
       setTimeout(() => {
         setPillActionIconName('copy-to-clipboard');
       }, 2000);
      setPillActionIconName('copy-to-clipboard-succeeded');
    }

```

Reply:
I'm using `useEffect` so we clean up with `return` if the component dismounts before the timeout is reached. React doc reference I used: https://react.dev/learn/synchronizing-with-effects#putting-it-all-together

Reply:
lmk if i'm misinterpreting anything

Reply:
This sounds good to me @garyghayrat